install/resources/upgrade34.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * MyBB 1.8
+ * Copyright 2014 MyBB Group, All Rights Reserved
+ *
+ * Website: http://www.mybb.com
+ * License: http://www.mybb.com/about/license
+ *
+ */
+
+/**
+ * Upgrade Script: 1.8.5
+ */
+
+$upgrade_detail = array(
+	"revert_all_templates" => 0,
+	"revert_all_themes" => 0,
+	"revert_all_settings" => 0
+);
+
+@set_time_limit(0);
+
+function upgrade34_dbchanges()
+{
+	global $db, $output;
+
+	$query = $db->simple_select('settings', 'value', "name='username_method'");
```

Comment:
> Why would I check a cached value instead of fetching it?

Because there is no reason to fetch it from the database, correct me if I'm wrong.

> It is an upgrade script not a production script.

That doesn't mean it shouldn't be optimal.

> I'm sure we do not check $mybb->settings when updating settings in any other version script.

https://github.com/mybb/mybb/blob/feature/install/resources/upgrade12.php#L800-814
We don't select them from db anywhere though.